gravitee-io/gravitee-access-management
# AM-76-mfa-rate-limit-gateway

This implementation of rate limit is applied for SMS and OTP factor. Current implementation is based on sliding token bucket algorithm.

## #: https://graviteecommunity.atlassian.net/browse/AM-76

### How to test

1. Uncomment `mfa_rate` section in gateway gravitee.yml and decrease the value for testing
2. Create SMS factor
3. In challenge page refresh the page so that it reaches `limit` value in gravitee.yml file
4. Now refresh again and you should get `mfa_request_limit_exceed` message.
